### PR TITLE
Fix None dataloader issue in PTL2.0

### PR DESCRIPTION
**Pull request by KunalDhawan**
# What does this PR do ?

With PTL2.0, returning None from the *_dataloader methods is no longer supported (reference - https://github.com/Lightning-AI/lightning/issues/17154). This change causes an issue with ASR decoding when we don't provide a valid trainer.validation_ds.manifest_filepath or trainer.test_ds.manifest_filepath. For example running the following code:
`python NeMo/examples/asr/asr_ctc/speech_to_text_ctc_bpe.py --config-path=../conf/citrinet/ --config-name=citrinet_1024.yaml model.tokenizer.dir=data/tokenizer/tokenizer_bpe_v1024/ model.tokenizer.type="bpe" model.train_ds.manifest_filepath=data/LibriSpeech/train_clean_100.json model.validation_ds.manifest_filepath=data/LibriSpeech/dev_clean.json model.train_ds.batch_size=16 +accelerator="gpu" +strategy="ddp" +devices=1 trainer.max_epochs=1 exp_manager.checkpoint_callback_params.monitor="train_loss"`

throws the error - 
`TypeError: An invalid dataloader was returned from EncDecCTCModelBPE.val_dataloader(). Found None.`

The following PR fixes this issue by making the default return value of val_dataloader and test_dataloader as `[]`

**Collection**: All

# Changelog 

```
  def val_dataloader(self):
      if self._validation_dl is not None:
          return self._validation_dl
      else: # None dataloader no longer supported in PTL2.0
          return []

  def test_dataloader(self):
      if self._test_dl is not None:
          return self._test_dl
      else: # None dataloader no longer supported in PTL2.0
          return []
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/core/classes/modelPT.py [hunk 1]
```diff
@@ -858,10 +858,14 @@ def train_dataloader(self):
     def val_dataloader(self):
         if self._validation_dl is not None:
             return self._validation_dl
+        else: # None dataloader no longer supported in PTL2.0
```
Comment (titu1994) [inline nemo/core/classes/modelPT.py]: Set self._validation_dl to [] as well

**Review by titu1994 — DISMISSED** [review]
Thanks for the fix !

File: nemo/core/classes/modelPT.py (hunk 1 above)
Comment (KunalDhawan) [reply nemo/core/classes/modelPT.py]: Fixed

**Review by titu1994 — APPROVED** [review]
Thanks !

**Pull request merged**